sympy/solvers/solvers.py

```diff
@@ -1557,7 +1557,10 @@ def _expand(p):
                         try:
                             t = Dummy('t')
                             iv = _solve(u - t, symbol, **flags)
-                            soln = list(ordered({i.subs(t, s) for i in iv for s in soln}))
+                            new_soln = list(ordered({i.subs(t, s) for i in iv for s in soln}))
```

Comment:
@smichr yes. 
the result for this equation is `[]` is apparently due to the fact that when `_solve(u-t)` results in `[-LambertW(-log(_t))/log(_t)]` and after substituting `1` it gives `[nan]` .
One thing can be done is to check if after substitution if it results `[nan]`, probably we should skip that substitution, and place the `root` instead. What do you say?

Also I am a bit confused in the logic that how substituting `soln` values in the result(`iv`) terms would get us the solution to the original solution.